sql/planbuilder/from.go

```diff
@@ -748,6 +748,12 @@ func (b *Builder) resolveView(name string, database sql.Database, asOf interface
 			b.parserOpts = sql.NewSqlModeFromString(viewDef.SqlMode).ParserOptions()
 			node, _, _, err := b.Parse(viewDef.TextDefinition, false)
 			if err != nil {
+				// TODO: Need better way to determine the view usage here.
+				//  Creating new view or updating existing view to reference non-existent table/view do not apply here.
+				//  Need to account for non-existing functions or users without appropriate privilege to the referenced table/column/function.
+				if (sql.ErrTableNotFound.Is(err) || sql.ErrColumnNotFound.Is(err)) && strings.HasPrefix(strings.ToLower(b.ctx.Query()), "select") {
```

Comment:
Good point about the error needing to be different for creating a new view with a bad column references, versus querying an existing view that has a bad column reference. Relying on the prefix match for `SELECT` works okay, but won't cover all cases and feels like there should be a better way... 

Does this code get invoked when creating a new view? It seems like it wouldn't, but I haven't tested. Any other ideas on how to avoid the string prefix matching check there?

Reply:
No, this error message is not returned when creating a new view referencing invalid table. This error message is returned only when querying **existing** view that its referenced tables, columns, functions or user privileges were altered to become invalid. 
Creating a new view or updating existing view to have invalid references to these will not cause this issue. 

I'll think of ideas to avoid prefix matching :)